Release 2.3.3 (#6220)
nautobot/nautobot: nautobot/docs/release-notes/version-2.3.md

As Django 3.2 has reached end-of-life, Nautobot 2.3 requires Django 4.2, the next long-term-support (LTS) version of Django. There are a number of changes in Django itself as a result of this upgrade; Nautobot App maintainers are urged to review the Django release-notes ([4.0](https://docs.djangoproject.com/en/4.2/releases/4.0/), [4.1](https://docs.djangoproject.com/en/4.2/releases/4.1/), [4.2](https://docs.djangoproject.com/en/4.2/releases/4.2/)), especially the relevant "Backwards incompatible changes" sections, to proactively identify any impact to their Apps.

<!-- towncrier release notes start -->
## v2.3.3 (2024-09-16)

### Fixed

> - [#6184](https://github.com/nautobot/nautobot/issues/6184) - Fixed string concatenation in `extras.models.groups._map_filter_fields` method.

Reviewer: Can we have a more user-meaningful note on this one?

Reviewer: ```suggestion
- [#6184](https://github.com/nautobot/nautobot/issues/6184) - Fixed an exception in `extras.models.groups._map_filter_fields` method when certain App `filter_extensions` were present.
```

> - [#6190](https://github.com/nautobot/nautobot/issues/6190) - Added `display` property to Prefix to display namespace along with the prefix to allow differentiation of prefixes in a UI selector

Reviewer: ```suggestion
- [#6190](https://github.com/nautobot/nautobot/issues/6190) - Added `display` property to Prefix to display its namespace along with the prefix to allow differentiation between prefixes in the UI.
```

> - [#6197](https://github.com/nautobot/nautobot/issues/6197) - Fixed view name split in `core.utils.lookup.get_model_for_view_name` function.

Reviewer: ```suggestion
- [#6197](https://github.com/nautobot/nautobot/issues/6197) - Fixed an exception in `core.utils.lookup.get_model_for_view_name` function when rendering certain App object list views.
```
